lib/sycamore/sycamore/docset.py

```diff
@@ -1028,23 +1031,59 @@ def threshold_filter(doc: Document, threshold) -> bool:
                 )[0]
                 doc.elements.sort(key=lambda e: e.properties.get(score_property_name, float("-inf")), reverse=True)
             evaluated_elements = 0
-            for element in doc.elements:
-                e_doc = Document(element.data)
-                if e_doc.field_to_value(field) is None:
-                    continue
-                e_doc = entity_extractor.extract_entity(e_doc)
-                element.properties[new_field] = e_doc.properties[new_field]
 
-                # todo: move data extraction and validation to entity extractor
-                score = int(re.findall(r"\d+", element.properties[new_field])[0])
-                # we're storing the element_index of the element that provides the highest match score for a document.
-                doc_source_field_name = f"{new_field}_source_element_index"
-                if score >= doc.get(doc_source_field_name, 0):
-                    doc.properties[f"{new_field}"] = score
-                    doc.properties[f"{new_field}_source_element_index"] = element.element_index
-                if score >= threshold:
-                    return True
-                evaluated_elements += 1
+            if tokenizer and max_tokens and field == "text_representation":
```

Comment:
I don't think you want custom logic for text_representation (a proxy of how data is stored in the document) here. I'd just use whichever property the function call requests. If every element is expected to have the same value for a given property, the caller is going to be setting use_elements to False and we won't get here